File: code-samples/Ruby/create_session_sample.rb
require "json"
require "net/http"
require "net/https"

class CreateSessionSample
  attr_accessor :system_id, :system_pwd, :user_name, :user_pwd

  def initialize
    @endpoint = "https://connect.gettyimages.com/v1/session/CreateSession"
    @system_id = "your system id"
    @system_pwd = "your system password"
    @user_name = "user name"
    @user_pwd = "user password"
  end

  def create_session
    request = {
        :RequestHeader =>
            {
                :Token => ""
            },
        :CreateSessionRequestBody =>
            {
                :SystemId => @system_id,
                :SystemPassword => @system_pwd,
                :UserName => @user_name,
                :UserPassword => @user_pwd
            }
    }

    response = post_json(request)

    #status = response["ResponseHeader"]["Status"]
    #token = response["CreateSessionResult"]["Token"]
    #secure_token = response["CreateSessionResult"]["SecureToken"]
  end

  def post_json(request)
    #You may wish to replace this code with your preferred library for posting and receiving JSON data.
    uri = URI.parse(@endpoint)
    http = Net::HTTP.new(uri.host, 443)
    http.use_ssl = true

    response = http.post(uri.path, request.to_json, {'Content-Type' =>'application/json'}).body
    JSON.parse(response)
  end
end

File: v2/code-samples/C#/createsessionsample.cs
// Requirements:
// - C# 4.0
// - Add System.Web.Extension.dll reference to your project so the code can compile.

using System.Collections.Generic;
using System.IO;
using System.Net;
using System.Web.Script.Serialization;
using System.Diagnostics;

namespace ConnectExample.Samples
{
    public class RequestHeader
    {
        public string Token { get; set; }
        public string CoordinationId { get; set; }
    }

    public class StatusEntry
    {
        public string Type { get; set; }
        public string Code { get; set; }
        public string Message { get; set; }
    }

    public class ResponseHeader
    {
        public string Status { get; set; }
        public ICollection<StatusEntry> StatusList { get; set; }
        public string CoordinationId { get; set; }
    }

    public class CreateSessionRequestBody
    {
        public string SystemId { get; set; }
        public string SystemPassword { get; set; }
        public string UserName { get; set; }
        public string UserPassword { get; set; }
    }

    public class CreateSessionTokenRequest
    {
        public RequestHeader RequestHeader { get; set; }
        public CreateSessionRequestBody CreateSessionRequestBody { get; set; }
    }

    public class CreateSessionResult
    {
        public string AccountId { get; set; }
        public string SecureToken { get; set; }
        public string Token { get; set; }
        public int TokenDurationMinutes { get; set; }
    }

    public class CreateSessionTokenResponse
    {
        public ResponseHeader ResponseHeader { get; set; }
        public CreateSessionResult CreateSessionResult { get; set; }
    }

    public class CreateSessionSample
    {
        public string GetToken()
        {
            return GetCreateSessionResponse().CreateSessionResult.Token;
        }

        public string GetSecureToken()
        {
            return GetCreateSessionResponse().CreateSessionResult.SecureToken;
        }

        public CreateSessionTokenResponse GetCreateSessionResponse()
        {
            const string createSessionRequestUrl = "https://connect.gettyimages.com/v1/session/CreateSession";
            var createSessionRequestBody = new CreateSessionRequestBody
            {
                SystemId = "systemId",
                SystemPassword = "systemPassword",
                UserName = "username",
                UserPassword = "password",
            };

            var createSessionTokenRequest = new CreateSessionTokenRequest
            {
                RequestHeader = new RequestHeader(),
                CreateSessionRequestBody = createSessionRequestBody
            };

            return MakeWebRequest(createSessionRequestUrl, createSessionTokenRequest);            
        }

        //You may wish to replace this code with your preferred library for posting and receiving JSON data.
        private static CreateSessionTokenResponse MakeWebRequest(string requestUrl, CreateSessionTokenRequest request)
        {
            var webRequest = WebRequest.Create(requestUrl) as HttpWebRequest;

            webRequest.Method = "POST";
            webRequest.ContentType = "application/json";

            var jsonSerializer = new JavaScriptSerializer();
            var requestStr = jsonSerializer.Serialize(request);

            Debug.WriteLine(requestStr);

            using (var writer = new StreamWriter(webRequest.GetRequestStream()))
            {
                writer.Write(requestStr);
                writer.Close();
            }

            var response = webRequest.GetResponse() as HttpWebResponse;

            string jsonResult;
            using (var reader = new StreamReader(response.GetResponseStream()))
            {
                jsonResult = reader.ReadToEnd();
                reader.Close();
            }

            Debug.WriteLine(jsonResult);

            return jsonSerializer.Deserialize<CreateSessionTokenResponse>(jsonResult);
        }

    }
}